src/tracks/playabletrack/wavetrack/ui/WaveClipAdjustBorderHandle.cpp

```diff
@@ -437,37 +487,62 @@ UIHandlePtr WaveClipAdjustBorderHandle::HitAnywhere(
         else if (std::abs(px - clipRect.GetRight()) <= BoundaryThreshold * 2)
            leftClip = clip;
     }
-
-    std::unique_ptr<AdjustPolicy> adjustPolicy;
+    
+    std::shared_ptr<WaveClip> adjustedClip;
+    bool adjustLeftBorder {false};
     if (leftClip && rightClip)
     {
-       //between adjacent clips
-       if(ClipParameters::GetClipRect(*leftClip, zoomInfo, rect).GetRight() > px)
-         adjustPolicy = std::make_unique<AdjustClipBorder>(waveTrack, leftClip, false, zoomInfo);//right border
-       else
-         adjustPolicy = std::make_unique<AdjustClipBorder>(waveTrack, rightClip, true, zoomInfo);//left border
+        //between adjacent clips
+        if(ClipParameters::GetClipRect(*leftClip, zoomInfo, rect).GetRight() > px)
+        {
+            adjustedClip = leftClip;
+            adjustLeftBorder = false;
+        }
+        else
+        {
+            adjustedClip = rightClip;
+            adjustLeftBorder = true;
+        }
     }
     else
     {
-       auto clip = leftClip ? leftClip : rightClip;
-       if (clip)
+       adjustedClip = leftClip ? leftClip : rightClip;
+       if (adjustedClip)
        {
           //single clip case, determine the border,
           //hit testing area differs from one
           //used for general case
-          auto clipRect = ClipParameters::GetClipRect(*clip.get(), zoomInfo, rect);
+          const auto clipRect = ClipParameters::GetClipRect(*adjustedClip, zoomInfo, rect);
           if (std::abs(px - clipRect.GetLeft()) <= BoundaryThreshold)
-             adjustPolicy = std::make_unique<AdjustClipBorder>(waveTrack, clip, true, zoomInfo);
+             adjustLeftBorder = true;
           else if (std::abs(px - clipRect.GetRight()) <= BoundaryThreshold)
-             adjustPolicy = std::make_unique<AdjustClipBorder>(waveTrack, clip, false, zoomInfo);
+             adjustLeftBorder = false;
+          else
+             adjustedClip.reset();
        }
     }
 
-    if(adjustPolicy)
-      return AssignUIHandlePtr(
-         holder,
-         std::make_shared<WaveClipAdjustBorderHandle>(adjustPolicy)
-      );
+    if(adjustedClip)
+    {
+       const auto isStretchMode = state.state.ControlDown();
```

Comment:
Important : The modifier agreed upon with @LWinterberg and @Tantacrul was `ALT`, in the end.